packages/types/index.ts

```diff
@@ -0,0 +1,161 @@
+   */
+  canContainUndefs: boolean;
+}
+
+/**
+ * Query operation output for quads.
+ * For example: SPARQL CONSTRUCT results
+ */
+export interface IActorQueryOperationOutputQuads extends IActorQueryOperationOutputStream {
+  /**
+   * The type of output.
+   */
+  type: 'quads';
+  /**
+   * The stream of quads.
+   */
+  quadStream: RDF.Stream & AsyncIterator<RDF.Quad>;
+}
+
+/**
+ * Query operation output for quads.
+ * For example: SPARQL ASK results
+ */
+export interface IActorQueryOperationOutputBoolean extends IActorQueryOperationOutputBase {
+  /**
+   * The type of output.
+   */
+  type: 'boolean';
+  /**
+   * A promise resolving to the boolean output of the operation.
+   */
+  booleanResult: Promise<boolean>;
+
+}
+
+/**
+ * Binds a quad pattern term's position to a variable.
+ */
+export type PatternBindings = Record<string, RDF.Variable>;
+
+export interface IActionQueryOperation extends IAction {
+  /**
+   * The query operation to handle.
+   */
+  operation: Algebra.Operation;
+}
+
+/**
+ * Query operation output.
+ * @see IActorQueryOperationOutputBindings, IActorQueryOperationOutputQuads, IActorQueryOperationOutputBoolean
+ */
+export type ActorQueryOperationOutput =
+  IActorQueryOperationOutputStream |
+  IActorQueryOperationOutputQuads |
+  IActorQueryOperationOutputBoolean;
+export interface IActorQueryOperationOutputBase {
+  /**
+   * The type of output.
+   */
+  type: string;
+  /**
+   * The resulting action context.
+   */
+  context?: ActionContext;
+}
+
+/**
+ * Super interface for query operation outputs that represent some for of stream.
+ * @see IActorQueryOperationOutputBindings, IActorQueryOperationOutputQuads
+ */
+export interface IActorQueryOperationOutputStream extends IActorQueryOperationOutputBase {
+  /**
+   * Callback that returns a promise that resolves to the metadata about the stream.
+   * This can contain things like the estimated number of total stream elements,
+   * or the order in which the bindings appear.
+   * This callback can be invoked multiple times.
+   * The actors that return this metadata will make sure that multiple calls properly cache this promise.
+   * Metadata will not be collected until this callback is invoked.
+   */
+  metadata?: () => Promise<Record<string, any>>;
+}
+
+/**
+ * Data interface for the type of action.
+ */
+export interface IAction {
+
+  /**
+   * The optional input context that is passed through by actors.
+   */
+  context?: ActionContext;
+}
+
+/**
+ * An immutable key-value mapped context that can be passed to any (@link IAction}.
+ * All actors that receive a context must forward this context to any actor, mediator or bus that it calls.
+ * This context may be transformed before forwarding.
+ *
+ * Each bus should describe in its action interface which context entries are possible (non-restrictive)
+ * and expose a `KEY_CONTEXT_${ENTRY_NAME}` constant for easy reuse.
```

Comment:
I suspect some recent changes were lost. E.g. https://github.com/comunica/comunica/commit/12b9ee3e8e5bc2d0fadd662a3d6aeef838b87619#diff-cfc7a9b5296e9c5c4950a2fde534bebbe7053cc63ae511026bfa251f9e56a1b6

Reply:
I completely missed the updated version of this jsdoc comment in moving the `ActionContext` type to `@comunica/types` and then rebasing my branch onto `master`. I've now updated the comment in `@comunica/types`  to match the version in `master` and modified the latter to add the `@deprecated` flag. Should be done.